arch/RISCV/RISCVDisassembler.c

```diff
@@ -315,6 +315,24 @@ static void clear_MI_insn_detail(MCInst *MI)
   	return;
 }
 
+// mark the load/store instructions through the opcode.
+static void markLSInsn(MCInst *MI, uint32_t in)
+{
+	/* 
+	   I   ld 0000011 = 0x03
+	       st 0100011 = 0x23
+	   F/D ld 0000111 = 0x07
+	       st 0100111 = 0x27
+	*/
+#define MASK_LS_INSN 0x0000007f
+	uint32_t opcode = in & MASK_LS_INSN;
+	if (0 == (opcode ^ 0x03) || 0 == (opcode ^ 0x07) ||
+	    0 == (opcode ^ 0x23) || 0 == (opcode ^ 0x27))
+		MI->flat_insn->detail->riscv.need_effective_addr = 1;
```

Comment:
this looks like a bug too: detail mode must be on, or you get crash here.

Reply:
Thanks, get it.